crates/e2e/tests/e2e/liquidity.rs

```diff
@@ -200,6 +207,9 @@ fn create_zeroex_liquidity_orders(
         // fully covers execution costs
         maker_amount: order_creation.buy_amount.as_u128() * 3,
         taker_amount: order_creation.sell_amount.as_u128() * 2,
+        // makes 0x order partially fillable, but the amount is higher than the cowswap order to
+        // make sure the 0x order is not overfilled in the end of the e2e test
+        remaining_fillable_taker_amount: order_creation.sell_amount.as_u128() * 3 / 2,
```

Comment:
So far it seems to be working. Let's try as the last step to actually fill the order that amount on-chain to get 100% confidence in the support for partially filled orders.
Should be doable by just calling fill_or_kill_limit_order (like the encoded 0x interaction would).